### ISSUE TITLE
Autoscaling bare metal clusters

### ISSUE THREAD

**Issue by MasterScrat** [kubernetes/autoscaler#1060]
I am running my own bare metal cluster. I would like to autoscale it by turning physical nodes on and off.

I imagine this would be doable using IPMI.

**Comment by aleksandra-malinowska** [comment]
In general, to support Cluster Autoscaler in a new environment, you'll need to implement its cloud provider interface for that environment. This is described in more detail in #953.

I've no idea how doable it is given your setup, but if you decide to give it a go, be sure to let us know if it works! :)

**Comment by tbrodbeck** [comment]
I would like to /reopen this issue. Is there progress regarding the topic over the course of the years?

**Comment by docent-net** [comment]
I went this road, of understanding, how CA could be used for bare-metal clusters,  and my understanding was that it simply cannot.

CA works on the ASG/node-group principle - and on bare-metal we don't have ASGs/node-groups. I tried designing a virtual node-group, and a virtual "cloud client" for bare metal, but there were so many issues with this design, that I gave up.

I ended up creating an own cluster-bare-autoscaler (https://github.com/docent-net/cluster-bare-autoscaler). Not production ready as for now (2025-04), but should be soon. Already does, what is suppose to do, but has some limitations. 

Awaiting your input folks!
